File: tests/fixtures/kotlin/nested_classes.kt
package com.example.nested

class Outer(val x: Int) {
    inner class Inner(val y: Int) {
        fun sum(): Int = x + y
    }

    class Nested {
        fun greet(): String = "Hello from Nested"
    }

    enum class Status {
        ACTIVE,
        INACTIVE
    }
}

File: tests/fixtures/scala/nested_classes.scala
package com.example.nested

class Outer(val name: String) {
  class Inner(val value: Int) {
    def describe(): String = s"$name: $value"
  }

  object InnerCompanion {
    def create(v: Int): Inner = new Inner(v)
  }
}

object Container {
  class Item(val id: Int)
  object ItemFactory {
    def make(id: Int): Item = new Item(id)
  }
}
